packages/checklist-service/test/services/checklists/entries/add.test.js

```diff
@@ -1,38 +1,48 @@
-const proxyquire = require('proxyquire')
-const { test } = require('tap')
+const t = require('tap')
+const { v4: uuid } = require('uuid')
 
 const { userId, userRequestContext } = require('../../../fixtures')
 
-const received = {}
-const payload = { title: 'entry1', value: '' }
-const addEntryHandler = proxyquire(
-  '../../../../services/checklists/entries/add',
-  {
-    './entries': {
-      addEntry: params => {
-        received.addEntryParams = params
-        return received
-      }
+let addEntryParams = {}
+let addEntryReturnVal = {}
+const addEntryHandler = t.mock('../../../../services/checklists/entries/add', {
+  '../../../../services/checklists/entries/entries.js': {
+    addEntry: params => {
+      addEntryParams = { ...params }
+      return Promise.resolve(addEntryReturnVal)
     }
   }
-)
+})
+
+t.beforeEach(async () => {
+  addEntryParams = {}
+  addEntryReturnVal = {}
+})
 
-test('add entry handler creates entries for checklists', async t => {
+t.test('add entry handler creates entries for checklists', async t => {
+  const listId = uuid()
+  const payload = { title: 'entry1', value: 'some-value' }
   const event = {
     requestContext: userRequestContext,
     body: JSON.stringify(payload),
 
     pathParameters: {
-      id: '1234'
+      id: listId
     }
   }
 
+  addEntryReturnVal = {
+    entId: uuid(),
+    title: payload.title,
+    value: payload.value
+  }
+
   const result = await addEntryHandler.main(event)
 
-  t.equal(received.addEntryParams.userId, userId)
-  t.equal(received.addEntryParams.listId, event.pathParameters.id)
-  t.equal(received.addEntryParams.title, payload.title)
-  t.equal(received.addEntryParams.value, payload.value)
+  t.equal(addEntryParams.userId, userId)
+  t.equal(addEntryParams.listId, listId)
+  t.equal(addEntryParams.title, payload.title)
+  t.equal(addEntryParams.value, payload.value)
   t.equal(result.statusCode, 201)
-  t.end()
```

Comment:
remove use of `t.end()` since the test fns are async and are not required - they will end once the promise resolves/rejects